src/utils/requests/__tests__/BillRequests.test.js

```diff
@@ -0,0 +1,32 @@
+import mockGetBill from "../__mocks__/BillRequests";
+import {getBill} from "../BillRequests"; 
+describe("BillRequests", () => {
+    const callGetBill = jest.fn(token => getBill(token));
```

Comment:
I don't understand what's going on here, can you explain? 

It seems to me you're mocking the getBill call into an unused constant. To me, it would make more sense if you mock the `fetch` api so that it could return what you want and then in this file you simply call `getBill` with the scenarios indicated. Creating the `mockGetBill` function call does not seem like the proper way. 

A link that might help you 

https://www.reactnativeschool.com/mocking-fetch-api-calls-when-using-jest

Reply:
You can also again refer to our previous code here: https://github.com/dannylekim/billsnapFront/pull/17/files#diff-483fbe43c486ff2cae5828bd7b7fb714

where they mocked the window.fetch api and still called the original function.

Reply:
its impossible to call the original function without a token, so the only response I would be able to test is the unauthorized response using a non valid token.

Reply:
What do you mean it's impossible to call the original function? It's a public function you simply call it (again keep in mind that you're supposed to be mocking the fetch call therefore NOT actually calling the actual backend)

Reply:
i cant call it and receive valid response besides unauthorized, because the bills route will always give unauthorized since no valid token is passed in test. in my original tests i put default tokens but you said its wrong.

Reply:
im only testing the unauthorized that means since its not possible to receive mock bills from getBill.

Reply:
It is possible to mock the fetch api as shown on the link. example:

`window.fetch.returns(mockApiResponse(goodResponse));`

I really suggest you taking a read at the PR here as I mentioned earlier to understand what I'm saying. There is nothing giving you an unauthorized response if you are not calling the actual backend.

Reply:
that doesnt work, it keeps throwing "window.fetch.returns is not a function", and I didnt want to import more dependencies in.

Reply:
> that doesnt work, it keeps throwing "window.fetch.returns is not a function", and I didnt want to import more dependencies in.

"sinon, and other stuff in userrequest"

Reply:
Okay, I'll take over this branch and try my hand, take a break you worked hard